### DISCUSSION TITLE
regarding libspdm_append_message_X

### DISCUSSION THREAD

**Discussion by zlyanz13** [DMTF/libspdm#860] [Q&A] [answered]
When processing unit test for responder certificate, 
function _libspdm_append_message_b_  is called and buffer for hash context is allocated.
But after that, it does not free the buffer which leads to memory leak.
Do I have to reset the buffer manually?

**Comment by steven-bellock** [comment, marked as answer]
So there is `libspdm_reset_context` but that just resets the context to be used again. Looks like we need a teardown function that would free any memory allocated by the cryptography library.
